### PR TITLE
add rustfmt support

### LINKED ISSUE #1906: [CLI] using archived crate: `gitignore.rs`
**Problem**

CLI using a archived crate: #1704 

**Screenshots**
image

**Environment:**
 - Dioxus version: `master`

**Questionnaire**
- [ ] I'm interested in fixing this myself but don't know where to start
- [X] I would like to fix and I have a solution
- [ ] I don't have time to fix this right now, but maybe later

### PR DESCRIPTION
**Pull request by Dangerised**
implement #1901

closes #1906

### CODE REVIEW

**Comment by ealmloff** [comment]
It looks like `cargo fmt` errors are getting captured somewhere without warning the user.

With this code:
```rust
fn main() {
```

Running `dx fmt --rustfmt` gives no output and does not format the code
Running `cargo fmt` gives this warning:
```
 --> /../testing/src/main.rs:1:12
  |
1 | fn main() {
  |           -^
  |           |
  |           unclosed delimiter
```

**Comment by Dangerised** [comment]
@ealmloff  

I've got some problems about `dx fmt`

dx-fmt

In the gif above , `dx` was built from crates.io, `dx-git` was built from github

also, I try to print out the files that `dx fmt` found, I found it empty.

image

**Comment by Dangerised** [comment]
- replace `gitignore` with `ignore` crate
- rewrite `autoformat_project` with rayon
- use `prettyplease` as formatter

**Comment by Dangerised** [comment]
fixed #1906

File: packages/cli/src/cli/autoformat.rs [hunk 1]
```diff
@@ -147,11 +149,13 @@ async fn format_file(
 ///
 /// Runs using Tokio for multithreading, so it should be really really fast
 ///
-/// Doesn't do mod-descending, so it will still try to format unreachable files. TODO.
-async fn autoformat_project(check: bool, split_line_attributes: bool) -> Result<()> {
-    let crate_config = dioxus_cli_config::CrateConfig::new(None)?;
-
-    let files_to_format = get_project_files(&crate_config);
+/// Doesnhttps://www.rustwiki.org.cn/zh-CN/std/'t do mod-descending, so it will still try to format unreachable files. TODO.
```
Comment (ealmloff) [inline packages/cli/src/cli/autoformat.rs:152]: Is the link to https://www.rustwiki.org.cn/zh-CN/std/ intentional?

File: packages/cli/src/error.rs [hunk 2]
```diff
@@ -46,6 +46,9 @@ pub enum Error {
 
     #[error(transparent)]
     Other(#[from] anyhow::Error),
+
+    #[error("Child process exited with {0}")]
```
Comment (ealmloff) [inline packages/cli/src/error.rs:50]: Now that this PR uses pretty please, we don't use this error variant

File: packages/cli/src/cli/autoformat.rs (hunk 1 above)
Comment (Dangerised) [reply packages/cli/src/cli/autoformat.rs:152]: no, by mistake

**Review by ealmloff — APPROVED** [review]
I pushed a few changes to make error reporting better and rename the rustfmt flag, other than that this looks good, thank you!

**Pull request merged**